src/Controller/User/AccountDeletionController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\User;
+
+use App\Controller\AbstractController;
+use App\Form\UserAccountDeletionType;
+use App\Service\IpResolver;
+use App\Service\UserManager;
+use Psr\Log\LoggerInterface;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\TooManyRequestsHttpException;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\RateLimiter\RateLimiterFactory;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class AccountDeletionController extends AbstractController
+{
+    public function __construct(
+        private readonly Security $security,
+        private readonly UserManager $userManager,
+        private readonly RateLimiterFactory $userDeleteLimiter,
+        private readonly IpResolver $ipResolver,
+        private readonly LoggerInterface $logger,
+        private readonly UserPasswordHasherInterface $userPasswordHasher,
+        private readonly TranslatorInterface $translator,
+    ) {
+    }
+
+    #[IsGranted('ROLE_USER')]
+    public function __invoke(Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('edit_profile', $this->getUserOrThrow());
+
+        $form = $this->createForm(UserAccountDeletionType::class);
+        $user = $this->getUserOrThrow();
+        try {
+            // Could throw an error
+            $form->handleRequest($request);
+            if ($form->isSubmitted() && $form->has('currentPassword')) {
+                if (!$this->userPasswordHasher->isPasswordValid(
+                    $this->getUser(),
```

Comment:
it looks like you already grab the user on ln 41, could this be changed to reuse that?

```suggestion
                    $user
```